pkg/services/ngalert/schedule/schedule.go

```diff
@@ -651,9 +639,27 @@ func (r *alertRuleRegistry) keyMap() map[models.AlertRuleKey]struct{} {
 }
 
 type alertRuleInfo struct {
-	evalCh  chan *evalContext
-	stopCh  chan struct{}
-	version int64
+	evalCh chan *evalContext
+	ctx    context.Context
+	Stop   context.CancelFunc
```

Comment:
Why is Stop exported here? Only remaining question I have beyond what George has already asked.

Reply:
No reason. I will change